### PR TITLE
Add mypy type checking.

### PR DESCRIPTION
**Pull request by jmchilton**
Various code cleanups (dropping Python 2 compat. mostly) to be enable this.

### CODE REVIEW

**Comment by bgruening** [comment]
:1st_place_medal: this is awesome. @jmchilton can you outline how you want to proceed with mypy maybe we can help.

**Comment by jmchilton** [comment]
I don't know 🤷‍♂️ - this isn't a real priority just something I was hacking on between rounds of fighting with CI on actually important PRs.

I think what is here right now is okay and will catch a few errors here and there and will encourage best practices to get us ready for typing. If I had to guess on another big next task it would probably be to include type annotations with galaxy-util, galaxy-tool-util, and bioblend. I had to disable import checking because a variety of libraries don't support it - so starting at the bottom of the dependency chain and working up would probably be the next step?

File: planemo/runnable.py [hunk 1]
```diff
@@ -6,8 +6,8 @@
 import collections
 import os
 from distutils.dir_util import copy_tree
+from enum import auto, Enum
```
Comment (nsoranzo) [inline planemo/runnable.py:9]: @jmchilton `enum.auto` is available only on Python >=3.6 (in fact CircleCI builds are broken). Should we remove Python 3.5 support? It reached EOL on 2020-09-30.

Comment (jmchilton) [reply planemo/runnable.py:9]: Yeah - go for it!

Comment (nsoranzo) [reply planemo/runnable.py:9]: Addressed in https://github.com/galaxyproject/planemo/pull/1099

**Pull request merged**